packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -101,30 +104,42 @@ export const createAddressTransactionsProvider = ({
                     `Last stored tx: ${lastStoredTransaction?.id} block:${lastStoredTransaction?.blockHeader.blockNo}`
                   );
 
-                const sinceBlock = lastStoredTransaction?.blockHeader.blockNo;
-                const newTransactions = await chainHistoryProvider.transactionsByAddresses({
-                  addresses,
-                  sinceBlock
-                });
+                const lowerBound = lastStoredTransaction?.blockHeader.blockNo;
+                let startAt = 0;
+                let response: Cardano.TxAlonzo[] = [];
+                let pageResults: Cardano.TxAlonzo[] = [];
+                do {
+                  pageResults = (
+                    await chainHistoryProvider.transactionsByAddresses({
+                      addresses,
+                      blockRange: { lowerBound },
+                      pagination: { limit: PAGE_SIZE, startAt }
+                    })
+                  ).pageResults;
+
+                  startAt += PAGE_SIZE;
+                  response = [...response, ...pageResults];
+                } while (pageResults.length === PAGE_SIZE);
```

Comment:
This is already a large function, please hoist this to a separate function. Also `newTransactions` is a little bit more descriptive than `response`.
```suggestion
                const newTransactions = await allTransactionsByAddresses(chainHistoryProvider, {addresses, blockRange: { lowerBound }})
```